libyul/optimiser/StackCompressor.h

```diff
@@ -27,13 +27,15 @@ namespace yul
 {
 
 struct Dialect;
-struct Block;
+struct Object;
 struct FunctionDefinition;
 
 /**
  * Optimisation stage that aggressively rematerializes certain variables in a function to free
  * space on the stack until it is compilable.
  *
+ * Only runs on the code of the object itself, does not descend into sub-objects.
```

Comment:
Is this behavior different from what was there before?

Reply:
It doesn't look like it, was the comment added just to be more explicit?

Reply:
The behaviour is the same, but we are now passing a full object containing multiple ASTs instead of just a single AST. Because of that, I added the explanation that in only considers the main ast inside the object.